go/chat/sourceofflinable.go

```diff
@@ -55,20 +56,26 @@ func (s *sourceOfflinable) IsOffline(ctx context.Context) bool {
 	s.Lock()
 	offline := s.offline
 	connected := s.connected
+	delayed := s.delayed
 	s.Unlock()
 
 	if offline {
-		select {
-		case <-connected:
-			s.Lock()
-			defer s.Unlock()
-			s.Debug(ctx, "IsOffline: waited and got %v", s.offline)
-			return s.offline
-		case <-time.After(4 * time.Second):
-			s.Lock()
-			defer s.Unlock()
-			s.Debug(ctx, "IsOffline: timed out")
-			return s.offline
+		if !delayed {
+			select {
+			case <-connected:
+				s.Lock()
+				defer s.Unlock()
+				s.Debug(ctx, "IsOffline: waited and got %v", s.offline)
+				return s.offline
+			case <-time.After(4 * time.Second):
+				s.Lock()
+				defer s.Unlock()
+				s.delayed = true
+				s.Debug(ctx, "IsOffline: timed out")
+				return s.offline
+			}
+		} else {
+			s.Debug(ctx, "IsOffline: offline, but skipping delay since we already did it")
```

Comment:
i think you still need a `return s.offline`?

Reply:
It just falls though to return it

Reply:
Ok, github diff fail on my part then...